internal/config/reminder/recurrence.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reminder
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/stacklok/minder/internal/config"
+)
+
+// RecurrenceConfig contains the configuration for the reminder recurrence
+type RecurrenceConfig struct {
+	// Interval is the time between reminders
+	Interval time.Duration `mapstructure:"interval" default:"1h"`
+	// BatchSize is the number of reminders to process at once. Batch size cannot be less than
+	// MaxPerProject * MinProjectFetchLimit.
+	BatchSize int `mapstructure:"batch_size" default:"100"`
+	// MaxPerProject is the maximum number of reminders per project in a batch
+	MaxPerProject int `mapstructure:"max_per_project" default:"10"`
+	// MinProjectFetchLimit is the minimum number of projects to fetch in an iteration. Additional
+	// projects are fetched if there is still space in the batch.
+	MinProjectFetchLimit int `mapstructure:"min_project_fetch_limit" default:"10"`
+	// MinElapsed is the minimum time after last update before sending a reminder
+	MinElapsed time.Duration `mapstructure:"min_elapsed" default:"1h"`
+}
+
+// InvalidBatchSizeError is a custom error type for the case when batch_size is less than
+// max_per_project * min_project_fetch_limit
+type InvalidBatchSizeError struct {
+	BatchSize            int
+	MaxPerProject        int
+	MinProjectFetchLimit int
+}
+
+func (e *InvalidBatchSizeError) Error() string {
+	return fmt.Sprintf("batch_size %d cannot be less than max_per_project(%d)*min_project_fetch_limit(%d)=%d",
+		e.BatchSize, e.MaxPerProject, e.MinProjectFetchLimit, e.MaxPerProject*e.MinProjectFetchLimit)
+}
+
+// Validate checks that the recurrence config is valid
+func (r RecurrenceConfig) Validate() error {
+	if r.MinElapsed < 0 {
+		return fmt.Errorf("min_elapsed %s cannot be negative", r.MinElapsed)
+	}
+
+	if r.Interval < 0 {
+		return fmt.Errorf("interval %s cannot be negative", r.Interval)
+	}
+
+	if r.BatchSize < r.MaxPerProject*r.MinProjectFetchLimit {
+		return &InvalidBatchSizeError{
+			BatchSize:            r.BatchSize,
+			MaxPerProject:        r.MaxPerProject,
+			MinProjectFetchLimit: r.MinProjectFetchLimit,
+		}
+	}
+
+	return nil
+}
```

Comment:
I feel like this would be too much patching. If the time specified is `< 0`, it's very fair to give an error. If we patch every user mistake, bad configs might creep in git repos, build scripts, etc. An error would force the user to fix it. wdyt?

Reply:
I'm willing to respect that.  We'll go with validation here.